### DISCUSSION TITLE
CMAKE build failure because grpc

### DISCUSSION THREAD

**Discussion by dimitrisfinas** [open-telemetry/opentelemetry-cpp#1210] [Q&A] [answered]
I tried to build opentelemetry-cpp (v1.2.0) with cmake following instructions: https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md and encountered error “CMake Error at /usr/local/lib/lib/cmake/grpc/gRPCConfig.cmake:13 (include):
 include could not find load file:
  /usr/local/lib/lib/cmake/grpc/gRPCTargets.cmake”. 

I think I followed all prerequisites, I am on ubuntu: bionic (18.04) and I am building grpc 1.39.1 on same container before building opentelemetry.
Did you already encounter this issue? Any solution or workaround it?

**Comment by lalitb** [comment]
This looks to be the problem with gRPC installation. Are you able to build a simple (non-opentelemetry) client-server example after installation as steps [here ](https://grpc.io/docs/languages/cpp/quickstart/)

**Reply by dimitrisfinas** [reply]
indeed, grpc is complex to build due to its dependencies. I found a solution to this issue by using another docker image (as I used multi-stage build in docker). see https://github.com/open-telemetry/opentelemetry-cpp/discussions/1229 formore details

**Comment by dimitrisfinas** [comment, marked as answer]
The full answer to this issue was provided by using docker image and scripts setup_grpc.sh as stated in discussion #1229
